esp-hal-embassy/src/executor/thread.rs

```diff
@@ -123,7 +123,7 @@ This will use software-interrupt 3 which isn't available for anything else to wa
 
         // we do not care about race conditions between the load and store operations,
         // interrupts will only set this value to true.
-        if SIGNAL_WORK_THREAD_MODE[cpu].load(Ordering::SeqCst) {
+        if SIGNAL_WORK_THREAD_MODE[cpu].load(Ordering::Acquire) {
```

Comment:
Acquire ordering doesn't make sense here - there is no matching Release store to define a happens-before relationship with. However, for some inexplicable reason, this measures to be slightly faster than Relaxed. Manually adding a `memw` instruction before a Relaxed load is significantly worse, so the change with this ordering is likely non-local.